subgraphs/gmx/src/entities/token.ts

```diff
@@ -0,0 +1,117 @@
+import { Address, BigDecimal, Bytes, ethereum } from "@graphprotocol/graph-ts";
+import { ERC20 } from "../../generated/GlpManager/ERC20";
+import { Token, RewardToken } from "../../generated/schema";
+import { getUsdPricePerToken } from "../prices";
+import {
+  BIGDECIMAL_ZERO,
+  RewardTokenType,
+  GLP_SYMBOL,
+  PRICE_CACHING_BLOCKS,
+  USDG_SYMBOL,
+  BIGDECIMAL_ONE,
+  INT_ZERO,
+} from "../utils/constants";
+
+export function getOrCreateToken(
+  event: ethereum.Event,
+  tokenAddress: Address
+): Token {
+  let token = Token.load(tokenAddress);
+
+  if (!token) {
+    token = new Token(tokenAddress);
+
+    token.name = fetchTokenName(tokenAddress);
+    token.symbol = fetchTokenSymbol(tokenAddress);
+    token.decimals = fetchTokenDecimals(tokenAddress) as i32;
+    token.lastPriceUSD = BIGDECIMAL_ZERO;
+    token.lastPriceBlockNumber = event.block.number;
+    token.save();
+  }
+
+  // GLP price will be computed elsewhere.
+  if (token.symbol == GLP_SYMBOL) {
```

Comment:
let's compare using token address. Symbols can be duplicated

Reply:
Fixed at b5d5ed81355ebf1ff84a231c0fa5aba7df61acf0 Thanks!